### ISSUE TITLE
[BUG] OCRmyPDF never finishes processing a scanned document

### ISSUE THREAD

**Issue by tedtms** [jonaswinkler/paperless-ng#1045]
**Description**
OCRmyPDF appears to never complete the processing of a scanned document. 

Paperless will attempt to consume the document but syslog shows an Error after a few seconds where the process dies and is "reincarnated" with the document never being completely processed. Paperless.log shows the document successfully detected with consumption happening and the document starting to be processed by OCRmyPDF. However, the process never completes. 

Behavior does NOT occur when uploading or consuming a document with text already present. OCRmyPDF is skipped, and the document is processed normally.

**To Reproduce**
1. Scan a document using a desktop scanner with auto-OCR disabled
2. Upload the document to paperless by whatever preferred method
3. Observe paperless-consumer detect and consume the document
4. Observe paperless-consumer detect the absence of text and hand off the document to OCRmyPDF
5. Observe OCRmyPDF never complete processing the document

**Screenshots/Logs**

syslog entry that occurs any time a document is consumed. process almost immediately dies and is reincarnated, but never completes. This error does not occur unless OCRmyPDF is called:
![image](https://user-images.githubusercontent.com/49801453/119194194-b7e01c80-ba50-11eb-9c61-18828a668bcc.png)

paperless.log entry. Document is given to OCRmyPDF but never progresses past this point:
![image](https://user-images.githubusercontent.com/49801453/119194880-bd8a3200-ba51-11eb-9e31-a6e6c61ed6eb.png)

**Troubleshooting steps performed**
- updating all dependencies
- upgrading from 1.4.2 to 1.4.4
- attempting upload via multiple methods
- restarting all paperless services
- uploading multiple, different documents
- rebooting machine

**Relevant information**
 - Ubuntu 20.04
 - paperless-ng v1.4.4
 - tesseract-ocr v4.1.1
 - Bare metal installation
 - PostgreSQL database
 - paperless.conf customizations:

```
PAPERLESS_REDIS=redis://localhost:6379
PAPERLESS_DBHOST=postgres.local
PAPERLESS_DBPORT=5432
PAPERLESS_DBNAME=[redacted]
PAPERLESS_DBUSER=[redacted]
PAPERLESS_DBPASS="[redacted]"

PAPERLESS_CONSUMPTION_DIR=/mnt/paperless/consume
PAPERLESS_DATA_DIR=/mnt/paperless/data
PAPERLESS_MEDIA_ROOT=/mnt/paperless/media
PAPERLESS_FILENAME_FORMAT={created_year}/{correspondent}/{title}

PAPERLESS_SECRET_KEY="[redacted]"
PAPERLESS_ALLOWED_HOSTS=[paperless IP],[reverse proxy IP],localhost

PAPERLESS_OCR_LANGUAGE=eng
PAPERLESS_OCR_MODE=skip_noarchive
PAPERLESS_OCR_OUTPUT_TYPE=pdfa
PAPERLESS_CONVERT_MEMORY_LIMIT=0

PAPERLESS_CONSUMER_POLLING=1
PAPERLESS_OPTIMIZE_THUMBNAILS=true
```

**Comment by jonaswinkler** [comment]
If you suspect that OCRmyPDF is at fault here, try using OCRmyPDF on that document standalone. The actual arguments are given in the logs. If the error persists, report at the OCRmyPDF repository.

**Comment by jonaswinkler** [comment]
I've seen you've provided an example file at the unpaper repo.

I've tried the file, and it works fine with my docker setup on 1.4.4.

This might really be a missing library on your end.

**Comment by tedtms** [comment]
Hmm. Well, I have all of the dependencies installed and updated that are listed on the paperless-ng Bare Metal setup article. How can I check all the libraries independently beyond that?

**Comment by tedtms** [comment]
OK. So, my use case necessitates using an external Postgres server, so I'm stuck using the baremetal installation method.

I did a full reinstallation on a fresh Ubuntu 20.04 machine, and I'm unfortunately getting the same exact results. I made sure to install all of the dependencies listed in the [setup documentation](https://paperless-ng.readthedocs.io/en/latest/setup.html#bare-metal-route), but the errors are the same. This output is taken directly from the paperless.log console in the web interface:

```
[2021-06-25 15:38:30,954] [INFO] [paperless.consumer] Consuming brother_05262021_070844_001132.pdf
[2021-06-25 15:38:30,974] [DEBUG] [paperless.consumer] Detected mime type: application/pdf
[2021-06-25 15:38:31,019] [DEBUG] [paperless.consumer] Parser: RasterisedDocumentParser
[2021-06-25 15:38:31,023] [DEBUG] [paperless.consumer] Parsing brother_05262021_070844_001132.pdf...
[2021-06-25 15:38:31,247] [DEBUG] [paperless.parsing.tesseract] Extracted text from PDF file /media/library/paperless/consume/brother_05262021_070844_001132.pdf
[2021-06-25 15:38:31,619] [DEBUG] [paperless.parsing.tesseract] Calling OCRmyPDF with args: {'input_file': '/media/library/paperless/consume/brother_05262021_070844_001132.pdf', 'output_file': '/tmp/paperless/paperless-8xearw5o/archive.pdf', 'use_threads': True, 'jobs': 1, 'language': 'eng', 'output_type': 'pdfa', 'progress_bar': False, 'skip_text': True, 'clean': True, 'deskew': True, 'rotate_pages': True, 'rotate_pages_threshold': 12.0, 'sidecar': '/tmp/paperless/paperless-8xearw5o/sidecar.txt'}
[2021-06-25 15:40:16,689] [DEBUG] [paperless.parsing.tesseract] Deleting directory /tmp/paperless/paperless-8xearw5o
[2021-06-25 15:40:17,062] [ERROR] [paperless.consumer] Error while consuming document brother_05262021_070844_001132.pdf: CalledProcessError: Command '['unpaper', '-v', '--dpi', '600.0', '--layout', 'none', '--mask-scan-size', '100', '--no-border-align', '--no-mask-center', '--no-grayfilter', '--no-blackfilter', '--no-deskew', '/tmp/tmptisvlqy9/input.pnm', '/tmp/tmptisvlqy9/output.ppm']' died with <Signals.SIGKILL: 9>.
Traceback (most recent call last):
  File "/opt/paperless/src/paperless_tesseract/parsers.py", line 241, in parse
    ocrmypdf.ocr(**args)
  File "/opt/paperless/.local/lib/python3.8/site-packages/ocrmypdf/api.py", line 340, in ocr
    return run_pipeline(options=options, plugin_manager=plugin_manager, api=True)
  File "/opt/paperless/.local/lib/python3.8/site-packages/ocrmypdf/_sync.py", line 374, in run_pipeline
    exec_concurrent(context, executor)
  File "/opt/paperless/.local/lib/python3.8/site-packages/ocrmypdf/_sync.py", line 271, in exec_concurrent
    executor(
  File "/opt/paperless/.local/lib/python3.8/site-packages/ocrmypdf/_concurrent.py", line 82, in __call__
    self._execute(
  File "/opt/paperless/.local/lib/python3.8/site-packages/ocrmypdf/builtin_plugins/concurrency.py", line 132, in _execute
    for result in results:
  File "/usr/lib/python3.8/multiprocessing/pool.py", line 868, in next
    raise value
  File "/usr/lib/python3.8/multiprocessing/pool.py", line 125, in worker
    result = (True, func(*args, **kwds))
  File "/opt/paperless/.local/lib/python3.8/site-packages/ocrmypdf/_sync.py", line 189, in exec_page_sync
    ocr_image, preprocess_out = make_intermediate_images(
  File "/opt/paperless/.local/lib/python3.8/site-packages/ocrmypdf/_sync.py", line 158, in make_intermediate_images
    ocr_image = preprocess(
  File "/opt/paperless/.local/lib/python3.8/site-packages/ocrmypdf/_sync.py", line 105, in preprocess
    image = preprocess_clean(image, page_context)
  File "/opt/paperless/.local/lib/python3.8/site-packages/ocrmypdf/_pipeline.py", line 486, in preprocess_clean
    unpaper.clean(
  File "/opt/paperless/.local/lib/python3.8/site-packages/ocrmypdf/_exec/unpaper.py", line 134, in clean
    run(input_file, output_file, dpi=dpi, mode_args=unpaper_args)
  File "/opt/paperless/.local/lib/python3.8/site-packages/ocrmypdf/_exec/unpaper.py", line 87, in run
    external_run(
  File "/opt/paperless/.local/lib/python3.8/site-packages/ocrmypdf/subprocess/__init__.py", line 51, in run
    proc = subprocess_run(args, env=env, **kwargs)
  File "/usr/lib/python3.8/subprocess.py", line 512, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['unpaper', '-v', '--dpi', '600.0', '--layout', 'none', '--mask-scan-size', '100', '--no-border-align', '--no-mask-center', '--no-grayfilter', '--no-blackfilter', '--no-deskew', '/tmp/tmptisvlqy9/input.pnm', '/tmp/tmptisvlqy9/output.ppm']' died with <Signals.SIGKILL: 9>.
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/opt/paperless/src/documents/consumer.py", line 248, in try_consume_file
    document_parser.parse(self.path, mime_type, self.filename)
  File "/opt/paperless/src/paperless_tesseract/parsers.py", line 290, in parse
    raise ParseError(f"{e.__class__.__name__}: {str(e)}")
documents.parsers.ParseError: CalledProcessError: Command '['unpaper', '-v', '--dpi', '600.0', '--layout', 'none', '--mask-scan-size', '100', '--no-border-align', '--no-mask-center', '--no-grayfilter', '--no-blackfilter', '--no-deskew', '/tmp/tmptisvlqy9/input.pnm', '/tmp/tmptisvlqy9/output.ppm']' died with <Signals.SIGKILL: 9>.
```
And attached is the offending document:
[brother_05262021_070844_001132.pdf](https://github.com/jonaswinkler/paperless-ng/files/6717467/brother_05262021_070844_001132.pdf)

**Comment by LynnAU** [comment]
Bump. Getting this error as well, but instead of running on bare metal I'm using the provided tika docker compose file. Running on Ubuntu 20.04.

I tried running the ocrmypdf directly on the input file and it exited with `ERROR - PriorOcrFoundError: page already has text! - aborting (use --force-ocr to force OCR)`. Which looks like it's confusing paperless-ng.

Log excerpt from paperless.log:
```
[2021-07-05 10:13:58,736] [INFO] [paperless.consumer] Consuming 2015-11-10-<snip>.pdf
[2021-07-05 10:13:58,740] [DEBUG] [paperless.consumer] Detected mime type: application/pdf
[2021-07-05 10:13:58,773] [DEBUG] [paperless.consumer] Parser: RasterisedDocumentParser
[2021-07-05 10:13:58,781] [DEBUG] [paperless.consumer] Parsing 2015-11-10-<snip>.pdf...
[2021-07-05 10:13:59,181] [DEBUG] [paperless.parsing.tesseract] Extracted text from PDF file /tmp/paperless/paperless-upload-fzcs8fz9
[2021-07-05 10:13:59,407] [DEBUG] [paperless.parsing.tesseract] Calling OCRmyPDF with args: {'input_file': '/tmp/paperless/paperless-upload-fzcs8fz9', 'output_file': '/tmp/paperless/paperless-zhfc2pv7/archive.pdf', 'use_threads': True, 'jobs': 1, 'language': 'eng', 'output_type': 'pdfa', 'progress_bar': False, 'skip_text': True, 'clean': True, 'deskew': True, 'rotate_pages': True, 'rotate_pages_threshold': 12.0, 'sidecar': '/tmp/paperless/paperless-zhfc2pv7/sidecar.txt'}
```

Logs from paperless-ng docker container:
```
10:10:13 [Q] INFO Process-1 created a task from schedule [Check all e-mail accounts]       
10:10:13 [Q] INFO Process-1:6 stopped doing work
10:10:13 [Q] INFO Processed [nevada-florida-cup-mississippi]
10:10:13 [Q] INFO recycled worker Process-1:6
10:10:13 [Q] INFO Process-1:7 ready for work at 350
10:13:58 [Q] INFO Process-1:7 processing [2015-11-10-<snip>.pdf]
10:13:58 [Q] INFO Enqueued 1
[2021-07-05 10:13:58,736] [INFO] [paperless.consumer] Consuming 2015-11-10-<snip>.pdf
10:14:01 [Q] ERROR reincarnated worker Process-1:7 after death
10:14:01 [Q] INFO Process-1:8 ready for work at 414
```

**Comment by tedtms** [comment]
So, I actually figured it out. I've been running paperless-ng in an Ubuntu 20.04 _proxmox LXC container_. For all intents and purposes, this behaves the same way as running a full-blown VM. Up until the most recent release(s), it had been running fine on 512MB of virtual memory. But by pure happenstance, I had my proxmox server's console monitor pulled up this morning and noticed an out of memory error on the main terminal:
```
Memory cgroup out of memory: Killed process 14395 (unpaper) total-vm:637688kB, anon-rss:371712kB, file-rss:0kB shmem-rss:0kB, UID:101104 pgtables:1160kB oom_score_adj:0
```
UID 101104 is how proxmox views my contianer's paperless user, and obviously unpaper is running on that container as well. For those unaware, LXC containers (and really, all others) use the main OS's kernel and abstract it for the containers. So these _kernel level_ out of memory errors were occurring on the main platform and not the actual container, so I never got an out of memory container _inside_ my paperless-ng instance.

After seeing this error, I upped my container's virtual memory to 1GB, and re-consumed a document. Lo and behold, it consumed without issue. So! I would highly recommend that anyone experiencing this error look at upping either their machine's or their container's memory. :)
